### PR TITLE
Implement uuid() basics

### PR DESCRIPTION
**Pull request by justinvdm**
Adds basic support for generating strings representing uuids.

## Important notes
* I've used [v5 uuids](https://en.wikipedia.org/wiki/Universally_unique_identifier#Versions_3_and_5_(namespace_name-based)) here instead of v4, since v4 is meant to be random (not what we're doing) and v5 is meant to map to some input value (what we're doing). One gotcha here, is that v5 uuids also need a namespace identifier - we just define a global one (maybe we can make the namespace an option at some point).
* The good news about v5 uuids is the security they offer:
> Version-3 and version-5 UUIDs have the property that the same namespace and name will map to the same UUID. However, neither the namespace nor name can be determined from the UUID, even if one of them is specified, except by brute-force search

@peterp ^ that makes me wonder about PII with copycat in general - I have no idea what guarantees fictional offers here - it might be technically possible for people to infer the input values from the output values given by copycat. That depends on what [string-hash's](https://www.npmjs.com/package/string-hash) algorithm offers. I think we might be able to guarantee this for everything provided by fictional if we get a guarantee that the hash is not reversible, I guess we'd want that, right?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,21 +10,23 @@
   "license": "MIT",
   "dependencies": {
     "@faker-js/faker": "6.2.0",
-    "fictional": "0.4.5"
+    "fictional": "0.4.5",
+    "uuid": "8.3.2"
```
Comment (justinvdm) [inline package.json:14]: Seeing as this is a lib, should we use ranges rather than pinning so that people are able to get the later versions of transitive deps? Otherwise we could be the blocker for security upgrades, for e.g.

I opted for pinning since its what we've done so far, but thought would be good to discuss.

**Comment by peterp** [comment]
> that makes me wonder about PII with copycat in general - I have no idea what guarantees fictional offers here - it might be technically possible for people to infer the input values from the output values given by copycat. That depends on what [string-hash's](https://www.npmjs.com/package/string-hash) algorithm offers. I think we might be able to guarantee this for everything provided by fictional if we get a guarantee that the hash is not reversible, I guess we'd want that, right?

@justinvdm That would be amazing, and definitely should be something that we consider as a requirement.

**Review by peterp — APPROVED** [review]
Ship it!

File: package.json (hunk 1 above)
Comment (peterp) [reply package.json:14]: I think this comes down to trust, and whilst we can assume good actors, I think we should consider bad actors, and I would rather have a human make the decision here than semver. I know we lock down the VM that runs inside Snaplet, but I would just want to be overly cautious.

Comment (peterp) [reply package.json:14]: But, I think we should discuss this in realtime.

**Comment by justinvdm** [comment]
> > that makes me wonder about PII with copycat in general - I have no idea what guarantees fictional offers here - it might be technically possible for people to infer the input values from the output values given by copycat. That depends on what [string-hash's](https://www.npmjs.com/package/string-hash) algorithm offers. I think we might be able to guarantee this for everything provided by fictional if we get a guarantee that the hash is not reversible, I guess we'd want that, right?
> 
> @justinvdm That would be amazing, and definitely should be something that we consider as a requirement.

I've changed fictional to use md5 under the hood as part of hashing, which gives us this security: I think djb2 was already difficult to reverse. But I couldn't find enough information to indicate that we got this guarantee, and that string-hash's modification of the algorithm (the algorithm fictional actually uses) also had this same guarantee. So thought might us well add md5 to the mix to give us this guarantee.

Worth noting that we still use string-hash after the md5 hash so that we can get an integer value that isn't as large as md5 results in decimal form. This doesn't change the fact that results are irreversible, but it does mean collisions are probably more likely that with md5. 

In other words: while we now have results that are infeasible to reverse, one could technically find inputs that map to the same result. I don't think we need that kind of security though (it would be needed for things like hashing passwords), so I'm assuming for now this is fine.

From what I could find out online, collisions are still apparently rare with djb2 (string-hash is a modification of djb2), so I don't think we'll have a problem with the quality of results we generate either. It's more that people could probably find collisions if they tried hard enough, but I don't think this is something we need to try avoid. I'm just taking people's word for it though, I don't really know much about this stuff.
